src/lib/operators/table_scan/abstract_dereferenced_column_table_scan_impl.hpp

```diff
@@ -43,6 +43,40 @@ class AbstractDereferencedColumnTableScanImpl : public AbstractTableScanImpl {
 
   const std::shared_ptr<const Table> _in_table;
   const ColumnID _column_id;
+
+  template <typename IteratorType>
+  void _handle_search_results(IteratorType begin, IteratorType end, const ChunkID chunk_id, PosList& matches,
```

Comment:
I don't like the verb "handle". It doesn't really convey any information. Maybe add_rows_to_matches?